fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -161,7 +161,7 @@ private Object deepCopyIfNeeded(@Nonnull TypeRepository typeRepository,
             return null;
         }
 
-        if (fieldType.isPrimitive()) {
+        if (fieldType.isPrimitive() || fieldType instanceof Type.Enum) {
```

Comment:
Right, also, it seems there is a missing code path for handling `enum` fields in `deepCopyMessageIfNeeded`. To make sure this works as expected, I would suggest the following test:

```
enum Corpus {
  CORPUS_UNSPECIFIED = 0;
  CORPUS_UNIVERSAL = 1;
  CORPUS_WEB = 2;
  CORPUS_IMAGES = 3;
  CORPUS_LOCAL = 4;
  CORPUS_NEWS = 5;
  CORPUS_PRODUCTS = 6;
  CORPUS_VIDEO = 7;
}
message SearchRequest {
  string query = 1;
  int32 page_number = 2;
  int32 result_per_page = 3;
  Corpus corpus = 4;
}

case1: CORPUS_WEB against CORPUS_WEB

case2: 
  {"foo", 42, 100, CORPUS_LOCAL} against  {"foo", 42, 100, CORPUS_LOCAL}

another case for this message:

message SearchRequest {
  string query = 1;
  int32 page_number = 2;
  int32 result_per_page = 3;
  repeated Corpus corpus = 4;
}

 {"foo", 42, 100, [CORPUS_LOCAL]} against  {"foo", 42, 100, [CORPUS_LOCAL]}

```

I think the first case will hit this code path, but the second case will hit the (currently missing) handling of `enum` inside a message. The last case, will hit handling `enum` for repeated (array) fields.